### ISSUE TITLE
Import resolution issue with MaybeRef types in vue-query code

### ISSUE THREAD

**Issue by thraizz** [anymaniax/orval#1067]
Hi, I am running into invalid generated types for `MaybeRef` when generating vue-query code, the imports come from `@tanstack/vue-query/build/lib/types` which does not exist, they split CommonJS and ESM into either the folder `@tanstack/vue-query/build/legacy/types` or `@tanstack/vue-query/build/modern/types`.

I have created a sample repo here: https://github.com/thraizz/vitejs-vite-mm54xg

## What are the steps to reproduce this issue?
1. Create a vue3 + vite project that uses orval to generate vue-query code from an openapi.yml
2. Generate vue-query code
3. Typecheck the code and see that the import of vue-query `MaybeRef` type fails

## What happens?
When generating the vue-query code, the import `@tanstack/vue-query/build/lib/types` is used - this is undefined as of the current version, `5.8.4`. This leads to the following typescript error:
`Cannot find module '@tanstack/vue-query/build/lib/types' or its corresponding type declarations.

## What were you expecting to happen?
The type import should resolve.

## Any logs, error output, etc?
I have prepared a sample repository for this:
https://github.com/thraizz/vitejs-vite-mm54xg

## What versions are you using?

**Operating System:** macOS 14.1 (23B74), Node 21.1.0, TypeScript 5.2.2
**Package Version:** orval: ^6.20.0, vue-query: ^5.8.4, vue ^3.3.4
**Browser Version:** (does not apply)

**Comment by melloware** [comment]
I think @Maxim-Mazurok is working on this in https://github.com/anymaniax/orval/pull/931

**Comment by Maxim-Mazurok** [comment]
> I think @Maxim-Mazurok is working on this in #931

Hello, no, this issue seems to arise after updating vue-query to the latest which is outside of scope for #931, so PR with a fix is welcome. Not sure which to use, in my experience, it's been easier to make ESM work in CJS than to deal with other ESM-related issues, so I am migrating my projects back from ESM to CJS for now. Hope this helps, cheers!

**Comment by thraizz** [comment]
I can fix the typescript errors by downgrading to vue-query@^4.0.0 and then manually replace the generated
`import type { MaybeRef } from "@tanstack/vue-query/build/lib/types";` 
with
`import type { MaybeRef } from "node_modules/@tanstack/vue-query/build/lib/types";` 
but that does not really seem like the right way to go, right? 😸

**Comment by Maxim-Mazurok** [comment]
It appears that in our projects we're using `import type { MaybeRef } from "vue";`

Our version of orval is heavily patched, and we're on a mission to create proper PRs to get rid of those patches. Maybe even one of my PRs already has this fix included, I'm not sure.

You can find our best version of orval in `bww` branch of my [fork](https://github.com/Maxim-Mazurok/orval). See https://github.com/Maxim-Mazurok/orval/blob/7e842261774321b9c0d6f24c67759bf6cc502213/samples/vue-query/src/api/endpoints/petstoreFromFileSpecWithTransformer.ts#L19C1-L19C37

That branch is supposed to include all the fixes that we're applying, see branches `679-proper`, `679`, `857`, `858`, `930`, `943`. AS far as I remember they are named after respective issue IDs in the main orval repo.

TLDR, this will get fixed eventually. But feel free to fix it separately. None of my PRs main focus is to address this issue. It probably arose naturally when I updated some packages, etc.

**Comment by melloware** [comment]
Thanks @Maxim-Mazurok looking forward to your patches!
